ios/dtx_codec/decoder.go

```diff
@@ -24,22 +24,34 @@ func ReadMessage(reader io.Reader) (Message, error) {
 	result := readHeader(header)
 
 	if result.IsFragment() {
-
 		// the first part of a fragmented message is only a header indicating the total length of
 		// the defragmented message
-		if result.IsFirstFragment() {
-			// put in the header as bytes here
-			result.fragmentBytes = header
-			return result, nil
+		if !result.IsFirstFragment() {
+			return Message{}, NewOutOfSync(fmt.Sprintf("Wrong Fragment Index: %d/%d", result.FragmentIndex, result.Fragments))
 		}
-		// 32 offset is correct, the binary starts with a payload header
-		messageBytes := make([]byte, result.MessageLength)
-		_, err := io.ReadFull(reader, messageBytes)
-		if err != nil {
-			return Message{}, err
+
+		// Read all fragments and merge payloads under result.fragmentBytes
+		for i := 1; i < int(result.Fragments); i++ {
```

Comment:
Maybe @danielpaulus can answer this. Can there be other messages between fragments? Because with this implementation we expect that to be the case, and before that there was a separate fragment decoder (which would now be dead code I assume https://github.com/danielpaulus/go-ios/blob/main/ios/dtx_codec/channel.go#L148)

Reply:
> Can there be other messages between fragments?
I don't know but I had the same worry. Tried to create that situation by running parallel tests first time I created this solution. I haven't succeeded in producing such behavior. All fragments arrive sequentially, with no messages injected in between as far as I can tell.

Reply:
My understanding is Apple also relies on this serial behavior since multiple concurrent connections are already multiplexed at the lower IP layer, over different sockets.

Reply:
Updated the branch, utilizing existing Daniel's defragmenter already present in channel.go.